Build ufs_weather_model on Gaea-C5 + update cubed sphere gitmodules for perturbation/increments for cold starts #2271

DavidBurrows-NCO opened the pull request:
## Commit Queue Requirements:
- [X] Fill out all sections of this template.
- [X] All sub component pull requests have been reviewed by their code managers.
- [x] Run the full Intel+GNU RT suite (compared to current baselines) on either Hera/Derecho/Hercules
- [ ] Commit 'test_changes.list' from previous step
---
## Description:
* Updates modulefiles/ufs_gaea.intel.lua to compile ufs_weather_model on Gaea-C5

### Commit Message:
* UFSWM - Update modulefiles/ufs_gaea.intel.lua to compile ufs_weather_model on Gaea-C5
*   * FV3 - 
    * atmos_cubed_sphere - move the call to read_da_inc outside the if external_ic/restart logic

### Priority:
* Normal

## Git Tracking
### UFSWM:
* Closes #2268 
* Closes https://github.com/ufs-community/ufs-weather-model/pull/2271

Sub component Pull Requests:
- FV3:  https://github.com/NOAA-EMC/fv3atm/pull/832
- atmos_cubed_sphere:  https://github.com/NOAA-GFDL/GFDL_atmos_cubed_sphere/pull/340

### UFSWM Blocking Dependencies:
* None

## Changes
### Regression Test Changes (Please commit test_changes.list):
* No Baseline Changes.

### Input data Changes:
* None.

### Library Changes/Upgrades:
* No Updates
  
---
<!-- STOP!!! THE FOLLOWING IS FOR CODE MANAGERS ONLY. PLEASE DO NOT FILL OUT -->
## Testing Log:
- RDHPCS
  - [x] Hera
  - [x] Orion
  - [x] Hercules
  - [x] Jet
  - [x] Gaea
  - [x] Derecho
- WCOSS2
  - [x] Dogwood/Cactus
  - [ ] Acorn
- [x] CI
- [ ] opnReqTest (complete task if unnecessary)

DavidBurrows-NCO commented:
Hello..This PR is in support of https://github.com/NOAA-EMC/global-workflow/issues/2535. Please let me know how I should proceed with testing.  Since this PR will not affect other platforms, do I need to run RT on Hera/Derecho/Hercules? Thanks for your time!

jkbk2004 commented:
@DavidBurrows-NCO Develop branch builds and runs ok on gaea-c5. Can you explain why we need the code change of this PR? Any issue porting global workflow to gaea-c5?

BrianCurtis-NOAA commented:
> @DavidBurrows-NCO Develop branch builds and runs ok on gaea-c5. Can you explain why we need the code change of this PR? Any issue porting global workflow to gaea-c5?

@jkbk2004 Have you looked into @DavidBurrows-NCO changes to see if it works for the UFSWM on Gaea-C5?

jkbk2004 commented:
> > @DavidBurrows-NCO Develop branch builds and runs ok on gaea-c5. Can you explain why we need the code change of this PR? Any issue porting global workflow to gaea-c5?
> 
> @jkbk2004 Have you looked into @DavidBurrows-NCO changes to see if it works for the UFSWM on Gaea-C5?

jkbk2004 commented:
Sorry for accidentally closing/re-opening. Anyway, @BrianCurtis-NOAA @DavidBurrows-NCO the gaea modulefile of this pr is a lot clean. Let me throw a run on gaea. @RatkoVasic-NOAA can you take a look at this modulefile update for gaea?

BrianCurtis-NOAA commented:
> Sorry for accidentally closing/re-opening. Anyway, @BrianCurtis-NOAA @DavidBurrows-NCO the gaea modulefile of this pr is a lot clean. Let me throw a run on gaea. @RatkoVasic-NOAA can you take a look at this modulefile update for gaea?

One concerning things to me was the removal of the stack-python as it might cause issues with the abort_dep_tasks.py script.

DavidBurrows-NCO commented:
Good Morning...the previous module file didn't contain a load cmake, so you get the following error upon building:
```
David.Burrows@gaea56 07:58 ufsm $ ./build.sh 
UFS MODEL DIR: /gpfs/f5/epic/scratch/David.Burrows/ufs_fails_to_build_on_gaea/ufsm
cmake: /opt/cray/pe/gcc/10.3.0/snos/lib64/libstdc++.so.6: version `GLIBCXX_3.4.30' not found (required by cmake)
```
I added the module load cmake then did some clean up as well. The modulefile for gaea now resembles gfs_utils, ufs_utils, and upp.
@BrianCurtis-NOAA I was thinking the same about python.  I see a mix of python, stack-python, and no python in the other module files. I can add it back if necessary. 
Thanks everyone!

RatkoVasic-NOAA commented:
This changes worked for me. @BrianCurtis-NOAA  can you give me example so I can try to get failure with abort_dep_tasks.py

jkbk2004 commented:
> This changes worked for me. @BrianCurtis-NOAA can you give me example so I can try to get failure with abort_dep_tasks.py

Mostly with importing python ecflow package in https://github.com/ufs-community/ufs-weather-model/blob/develop/tests/abort_dep_tasks.py#L3. But it sounds like it runs ok. @RatkoVasic-NOAA a decision point on your side is to specifically control thru stack-python. I think its safe to keep moving with the stack-python. @DavidBurrows-NCO can you put the stack-python line back. So we can move on.

DavidBurrows-NCO commented:
@BrianCurtis-NOAA @RatkoVasic-NOAA I added stack-python load back into the module file and tested a build.

BrianCurtis-NOAA commented:
> This changes worked for me. @BrianCurtis-NOAA can you give me example so I can try to get failure with abort_dep_tasks.py

This concern is just based on rt.sh failing with running ECFlow on Gaea. @RatkoVasic-NOAA are you using ecflow to run the rt.sh suite?

RatkoVasic-NOAA commented:
@BrianCurtis-NOAA I didn't run full test suite, just about 10 test cases to see if it brakes compilation or some executions. I used rocoto. I can now try with ecflow same thing.

RatkoVasic-NOAA commented:
Selected 10 tests worked using both ecflow and rocoto for me.

BrianCurtis-NOAA commented:
> Selected 10 tests worked using both ecflow and rocoto for me.

OK Great. @jkbk2004 please run the full rt suite on Gaea-C5 using ecflow and we can then merge this PR with another non-baseline changing PR once that's completed.

jkbk2004 commented:
> > Selected 10 tests worked using both ecflow and rocoto for me.
> 
> OK Great. @jkbk2004 please run the full rt suite on Gaea-C5 using ecflow and we can then merge this PR with another non-baseline changing PR once that's completed.

Sure!

jkbk2004 commented:
@DavidBurrows-NCO This rap clm_lake case is crashing with this pr. experiment is at /gpfs/f5/epic/scratch/Jong.Kim/RT_RUNDIRS
/Jong.Kim/FV3_RT/rt_48937/rap_clm_lake_debug_intel/err. Develop runs ok. @SamuelTrahanNOAA just FYI

DavidBurrows-NCO commented:
@jkbk2004 I am also getting a failure with that test.  Here are some highlights from the output:
```

 0: PASS: fcstRUN phase 1, n_atmsteps =               11 time is        14.979133
  0:  ---isec,seconds        3600        3600
  0:   gfs diags time since last bucket empty:    1.00000000000000      hrs
  0:  write out restart at n_atmsteps=          11  seconds=        3600
  0:  integration length=  0.9166667
  0: PASS: fcstRUN phase 2, n_atmsteps =               11 time is         0.788611
  0:   aft fcst run output time=        3600 FBcount=           8 na=          12
...
  0:  liq_aero  3.4789200E+10  1.1008354E+07  1.7738830E+09
  0:  ice_aero  1.4927048E+08  1.2824743E-13   3184276.
  0:  sgs_tke   149.9711      9.9780387E-05   1.240300
  0: NOTE from PE     0: Potential error in diag_manager_end: slp NOT available, check if output interval > runlength. Netcdf fill_values are written
  0: NOTE from PE     0: Potential error in diag_manager_end: vort850 NOT available, check if output interval > runlength. Netcdf fill_values are written
  0: NOTE from PE     0: Potential error in diag_manager_end: vort200 NOT available, check if output interval > runlength. Netcdf fill_values are written
  0: NOTE from PE     0: Potential error in diag_manager_end: us NOT available, check if output interval > runlength. Netcdf fill_values are written
  0: NOTE from PE     0: Potential error in diag_manager_end: u1000 NOT available, check if output interval > runlength. Netcdf fill_values are written
...
144: Stack trace terminated abnormally.
145: forrtl: error (73): floating divide by zero
```

Are you getting the same error? I don't see any other hints in the working directory for a reason for this failure.

SamuelTrahanNOAA commented:
I cannot see your directories due to restrictive permissions.

What you've shown me so far are errors that have nothing to do with CLM lake. I cannot debug farther until I see the stack trace.

DavidBurrows-NCO commented:
@SamuelTrahanNOAA Can I move the working dir somewhere on Gaea for you?  My Gaea groups are epic and ufs-ard.
The only stack trace I see in the working dir is:
```
144: Stack trace terminated abnormally.
145: forrtl: error (73): floating divide by zero
```
Thanks, Sam!

SamuelTrahanNOAA commented:
> I move the working dir somewhere on Gaea for you?

I don't know where your working directory resides.

You don't necessarily need to move it; I just need the permissions to allow world access.

This is not a stack trace:

> 145: forrtl: error (73): floating divide by zero

The stack trace would include a trace of the files and line numbers to the point in the stack with the divide by zero.

DavidBurrows-NCO commented:
@SamuelTrahanNOAA The working dir is here: `/gpfs/f5/ufs-ard/scratch/David.Burrows/RT_RUNDIRS/David.Burrows/FV3_RT/rt_200821`.
perms for files there are: `-rw-r--r--`, so you should be good.

SamuelTrahanNOAA commented:
Every path component must be readable. Your personal directory here is only readable by you:

/gpfs/f5/ufs-ard/scratch/David.Burrows

The same is true of everyone else in ufs-ard. That is by design; the GFDL admins created this separate area for world-readable files:

/gpfs/f5/ufs-ard/world-shared/

If you want to share things with everyone on GAEA, it belongs in the world-shared directory.

There's also a proj-shared, readable only by the unix group that owns the directory (ufs-ard):

/gpfs/f5/ufs-ard/proj-shared/

DavidBurrows-NCO commented:
Thanks @SamuelTrahanNOAA for that succinct explanation of directory perm structure on Gaea. I now see that /gpfs/f5/ufs-ard/scratch/$USER is only readable by owner.
Hopefully this works: /gpfs/f5/ufs-ard/world-shared/for_sam/rt_200821

SamuelTrahanNOAA commented:
The write component ranks are all failing with "floating divide by zero." That means they're diving a floating-point real number by zero. It is odd that there was no stack trace. I wonder if this happened after MPI_Finalize? Some of my UPP jobs freeze during MPI_Finalize due to unknown system issues.

Before the "floating divide by zero," I see errors about the output interval being longer than the runlength for the atmos_4xdaily fields. Indeed it is.

Output interval from diag_table: `"atmos_4xdaily",           6,  "hours",    1, "days",  "time"`
Run length from model_configure: `nhours_fcst:             1`

Please try disabling the atmos_4xdaily output. (Increasing the run length to 6 hours would probably pass the wallclock limit.) Delete all lines from diag_table that contain the string `atmos_4xdaily`. That file is `tests/parm/diag_table/diag_table_rap` in the repository.

DavidBurrows-NCO commented:
Morning @SamuelTrahanNOAA
I tried your first suggestion, removing all atmos_4xdaily lines from `tests/parm/diag_table/diag_table_rap` but got the same failure...I put the working directory here: `/gpfs/f5/ufs-ard/world-shared/for_sam/rt_200821`

I reverted `tests/parm/diag_table/diag_table_rap` and updated `FHMAX` to 6 in `tests/rap_clm_lake_debug`. That ran past the point of failure but hit wall clock (as you thought). I put that working directory here: `/gpfs/f5/ufs-ard/world-shared/for_sam/rt_249597`

BrianCurtis-NOAA commented on modulefiles/ufs_gaea.intel.lua:
```diff
@@ -5,13 +5,7 @@ help([[
 
 whatis([===[Loads libraries needed for building the UFS Weather Model on Gaea ]===])
 
-load("PrgEnv-intel/8.3.3")
-load("intel-classic/2023.1.0")
-load("cray-mpich/8.1.25")
-load("python/3.9.12")
```
@SamuelTrahanNOAA these loads are removed from the modulefile for gaea, I would guess that one of these needs to be loaded still to avoid the errors you're seeing.

DavidBurrows-NCO replied:
I initially thought the same thing then checked the output for modules loaded (see below). Python is the only non-matching module. I'll try python/3.9.12 and report back...
```
Currently Loaded Modules:
  1) craype-x86-rome                        30) hdf5/1.14.0
  2) perftools-base/23.03.0                 31) snappy/1.1.10
  3) xpmem/2.6.2-2.5_2.38__gd067c3f.shasta  32) zstd/1.5.2
  4) cray-pmi/6.1.10                        33) c-blosc/1.21.4
  5) darshan-runtime/3.4.0                  34) curl/7.66.0
  6) CmrsEnv/default                        35) netcdf-c/4.9.2
  7) TimeZoneEDT/default                    36) netcdf-fortran/4.6.0
  8) DefApps/default                        37) parallel-netcdf/1.12.2
  9) cray-dsmml/0.2.2                       38) parallelio/2.5.10
 10) cray-libsci/23.02.1.1                  39) esmf/8.5.0
 **11) PrgEnv-intel/8.3.3**                     40) fms/2023.02.01
 **12) intel-classic/2023.1.0**                 41) bacio/2.4.1
 13) craype/2.7.20                          42) crtm-fix/2.4.0_emc
 14) stack-intel/2023.1.0                   43) git-lfs/2.11.0
 15) craype-network-ofi                     44) crtm/2.4.0
 **16) cray-mpich/8.1.25**                      45) g2/3.4.5
 17) stack-cray-mpich/8.1.25                46) g2tmpl/1.10.2
 18) gettext/0.20.2                         47) ip/4.3.0
 19) libxcrypt/4.4.35                       48) sp/2.3.3
 20) sqlite/3.42.0                          49) w3emc/2.10.0
 21) util-linux-uuid/2.38.1                 50) gftl/1.10.0
 **22) python/3.10.8**                          51) gftl-shared/1.6.1
 23) stack-python/3.10.8                    52) fargparse/1.5.0
 24) cmake/3.23.1                           53) mapl/2.40.3-esmf-8.5.0
 25) libjpeg/2.1.0                          54) scotch/7.0.4
 26) jasper/2.0.32                          55) ufs_common
 27) zlib/1.2.13                            56) nccmp/1.9.0.1
 28) libpng/1.6.37                          57) modules.fv3
 29) pkg-config/0.29.2
```

DavidBurrows-NCO replied:
`python/3.9.12` still produces the same original error.  For completeness I copied that working directory to: `/gpfs/f5/ufs-ard/world-shared/for_sam/rt_65400`

DusanJovic-NOAA commented on modulefiles/ufs_gaea.intel.lua:
```diff
@@ -22,11 +16,13 @@ load(pathJoin("stack-cray-mpich", stack_cray_mpich_ver))
 load("ufs_common")
-load("nccmp/1.9.0.1")
 
-unload("darshan-runtime")
-unload("cray-libsci")
```
Why are these removed?

SamuelTrahanNOAA replied:
I know we need to unload `cray-libsci`. The intel and cray libraries contain routines of the same name and using the wrong one can cause a program to freeze while finalizing libraries (ie. during atexit handlers). As for darshan-runtime, I don't know what that is.

Please try putting those two lines back in.

DavidBurrows-NCO replied:
Adding the unload darshan and cray-libsci back to the module file seemed to work: `/gpfs/f5/ufs-ard/world-shared/for_sam/rt_126272`.
We still get these notices:
`0: NOTE from PE     0: Potential error in diag_manager_end: slp NOT available, check if output interval > runlength. Netcdf fill_values are written,`
I updated the module file in the repo.

This is from Gaea docs about darshan:
When the darshan-runtime/3.4.0 module is active with non-Intel compilers, an error is thrown at link time regarding Intel specific libraries that can't be found. This is an ongoing issue and is currently being worked on by ORNL (https://gaeadocs.rdhpcs.noaa.gov/wiki/index.php?title=Known_issues&title=Known_issues)

We should probably clean up tests/rt.sh as well, since the gaea module file is handling the module loads.  I'm not sure how much of this is needed:
```
    export LD_PRELOAD=/opt/cray/pe/gcc/12.2.0/snos/lib64/libstdc++.so.6
    module load PrgEnv-intel/8.3.3
    module load intel-classic/2023.1.0
    module load cray-mpich/8.1.25
    module load python/3.9.12
    module use /ncrc/proj/epic/spack-stack/modulefiles
    module load gcc/12.2.0
```

SamuelTrahanNOAA replied:
> We still get these notices:
> `0: NOTE from PE     0: Potential error in diag_manager_end: slp NOT available, check if output interval > runlength. Netcdf fill_values are written,`

Those warnings are okay. Actually, it's probably best to leave that in as a test of whether the diag manager is correctly filling with fill_values.

DavidBurrows-NCO commented:
Morning, everyone... @SamuelTrahanNOAA @DusanJovic-NOAA @BrianCurtis-NOAA Thanks for suggestions as to possible reasons for the rap_clm_lake_debug_intel failure.  Adding the unload darshan and cray-libsci back to the module file seemed to work. @jkbk2004 @RatkoVasic-NOAA How should I proceed with testing? Thanks!

jkbk2004 commented:
> Morning, everyone... @SamuelTrahanNOAA @DusanJovic-NOAA @BrianCurtis-NOAA Thanks for suggestions as to possible reasons for the rap_clm_lake_debug_intel failure. Adding the unload darshan and cray-libsci back to the module file seemed to work. @jkbk2004 @RatkoVasic-NOAA How should I proceed with testing? Thanks!

@DavidBurrows-NCO we can combine with another no baseline change PR once will re-confirm.

DavidBurrows-NCO commented:
Thanks @jkbk2004. I am also working with @weihuang-jedi on general porting of global-workflow to Gaea and NOAA CSPs. Is there a way to combine [PR#2269](https://github.com/ufs-community/ufs-weather-model/pull/2269) with this PR? Just checking..

jkbk2004 commented:
> Thanks @jkbk2004. I am also working with @weihuang-jedi on general porting of global-workflow to Gaea and NOAA CSPs. Is there a way to combine [PR#2269](https://github.com/ufs-community/ufs-weather-model/pull/2269) with this PR? Just checking..

@DavidBurrows-NCO One thing to check... #2258 will come after we merge #2093.  Is the timeline ok?

weihuang-jedi commented:
I was thinking of letting #2093 merge first, and then #2258.
But during today's meeting, Rahul suggests that we merge these two
together, and there is so many thing are waiting.
I have tried clone branch sync with #2093 on Gaea, and trying to pull #2258
changes in, I'll let @David Burrows - NOAA Affiliate
***@***.***> take a look first, and then start discussing if we
can combine these two.


On Tue, May 14, 2024 at 9:22 AM JONG KIM ***@***.***> wrote:

> Thanks @jkbk2004 <https://github.com/jkbk2004>. I am also working with
> @weihuang-jedi <https://github.com/weihuang-jedi> on general porting of
> global-workflow to Gaea and NOAA CSPs. Is there a way to combine PR#2269
> <https://github.com/ufs-community/ufs-weather-model/pull/2269> with this
> PR? Just checking..
>
> @DavidBurrows-NCO <https://github.com/DavidBurrows-NCO> One thing to
> check... #2258
> <https://github.com/ufs-community/ufs-weather-model/pull/2258> will come
> after we merge #2093
> <https://github.com/ufs-community/ufs-weather-model/pull/2093>. Is the
> timeline ok?
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/ufs-community/ufs-weather-model/pull/2269#issuecomment-2110519574>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ASMCH63IGLTHRO66ST2MEPDZCIT33AVCNFSM6AAAAABHJNA2ECVHI2DSMVQWIX3LMV43OSLTON2WKQ3PNVWWK3TUHMZDCMJQGUYTSNJXGQ>
> .
> You are receiving this because you were mentioned.Message ID:
> ***@***.***>
>

DavidBurrows-NCO commented:
@jkbk2004 @weihuang-jedi I don't see an issue with the merge order. The only file in both PRs is `modulefiles/ufs_noaacloud.intel.lua` and both are pointing to `spack-stack` at:
`/contrib/spack-stack/spack-stack-1.6.0/envs/unified-env/install/modulefiles/Core`. Wei then adds a new file `modulefiles/ufs_common4noaacloud.lua` to handle common modules on CSPs and a few other small file changes.

junwang-noaa commented on modulefiles/ufs_common4noaacloud.lua:
May I ask why do we need this file? Is there any issue just using the ufs_common.lua? How do we coordinate syncing the two common modules and testing the model code?

weihuang-jedi replied:
We actually do not need this file now.
@DavidBurrows-NCO could you please deleted this file "ufs_common4noaacloud.lua"?
Thanks.

DavidBurrows-NCO replied:
@junwang-noaa @weihuang-jedi I have removed ufs_common4noaacloud.lua.

jkbk2004 commented:
@DavidBurrows-NCO can you sync up the branch? We can start working on this PR. We are considering to combine with #2271 as no baseline changes are involved.

DavidBurrows-NCO commented:
@jkbk2004 Just synced it...

jkbk2004 commented:
@DavidBurrows-NCO can you point to https://github.com/pjpegion/fv3atm/tree/cs_perts, which is the fv3 feature branch from #2271 ? There is conflict in gaea module file. Can you resolve the conflicts as well?

DavidBurrows-NCO commented:
@jkbk2004 I updated Gaea module file to 1.6.0 and resolved conflict...for pointing to cs_perts, can you give me an idea of how to do that? My team and I aren't 100% how to do it. Thanks, Jong.

jkbk2004 commented:
> @jkbk2004 I updated Gaea module file to 1.6.0 and resolved conflict...for pointing to cs_perts, can you give me an idea of how to do that? My team and I aren't 100% how to do it. Thanks, Jong.

@DavidBurrows-NCO Component code change instruction is here: https://github.com/ufs-community/ufs-weather-model/wiki/Making-code-changes-in-the-UFS-weather-model-and-its-subcomponents#developing-in-subcomponents. Basically, update gitmodules and sync and commit/push to origin feature branch. I am ok to directly update your feature branch. Do you want me to update?

DavidBurrows-NCO commented:
@jkbk2004 That's what I figured the process was...but if you can do it from your side, I'm okay with that. Please update.

BrianCurtis-NOAA commented:
What's going on with the AutoRT?

jkbk2004 commented:
@zach1221 @FernandoAndrade-NOAA There are quite some issues on jenkins: handling git labels, multiple commits of logs, functionality on jet/hercules/hera, etc. Please, follow up with Platform team. Anyway, I am manually running on hercules/hera/jet. @ankimball Note that we need troubleshooting for jenkins issues.

BrianCurtis-NOAA commented on tests/detect_machine.sh:
```diff
@@ -51,16 +51,14 @@ case $(hostname -f) in
 
   discover3[1-5].prv.cube) MACHINE_ID=discover ;; ### discover31-35
-  *) MACHINE_ID=UNKNOWN ;;  # Unknown platform
+  *)
+      case ${PW_CSP:-} in 
+         "aws" | "google" | "azure") MACHINE_ID=noaacloud ;;
+         *) PW_CSP="UNKNOWN"
+            MACHINE_ID=UNKNOWN ;;  # Unknown platform
+      esac 
```
@DavidBurrows-NCO Could you explain the need to move this into the case block for hostname?

DavidBurrows-NCO replied:
Hi @BrianCurtis-NOAA..it was done to consolidate all of the MACHINE_ID statements..Please let me know if we should revert. Thanks!

BrianCurtis-NOAA replied:
Since there is a need to use this detect_machine.sh in global-workflow and its sub components this change isn't necessary enough to make. Please revert.

DavidBurrows-NCO replied:
@BrianCurtis-NOAA That makes sense. I have just pushed the revert. Thanks

zach1221 commented:
> @zach1221 @FernandoAndrade-NOAA There are quite some issues on jenkins: handling git labels, multiple commits of logs, functionality on jet/hercules/hera, etc. Please, follow up with Platform team. Anyway, I am manually running on hercules/hera/jet. @ankimball Note that we need troubleshooting for jenkins issues.

Ok, @jkbk2004 . I'll look into the issues with platform. Note, autoRT isn't setup to run yet on Jet.

jkbk2004 commented:
@DavidBurrows-NCO looks like something screwed up with your last commit to revert.

DavidBurrows-NCO commented:
@jkbk2004 checking on this now

DavidBurrows-NCO commented:
@jkbk2004 Should I add the updated .gitmodules back to the repo too?

jkbk2004 commented:
@DavidBurrows-NCO git revert is very powerful. In this case, it did undo multiple commits. Anyway, it looks like it's safe to start from scratch again. We need to point to https://github.com/pjpegion/fv3atm/tree/cs_perts

jkbk2004 commented:
@DavidBurrows-NCO FYI: git cherry-pick <commit-id> can be used to "un-revert" a reverted Git commit, I think. But feel free to start from scratch.

jkbk2004 commented:
@DavidBurrows-NCO please, resolve conflicts in gaea lua file

jkbk2004 commented:
@BinLiu-NOAA I see a minor commit on HYCOM: https://github.com/NOAA-EMC/HYCOM-src/commit/bcf7777bb037ae2feb2a8a8ac51aacb3511b52d9. Do you want us to catch up the hash?

BinLiu-NOAA commented:
> @BinLiu-NOAA I see a minor commit on HYCOM: [NOAA-EMC/HYCOM-src@bcf7777](https://github.com/NOAA-EMC/HYCOM-src/commit/bcf7777bb037ae2feb2a8a8ac51aacb3511b52d9). Do you want us to catch up the hash?

Thanks @jkbk2004! Yes, if it can be combined with any PR like this one. It is just a cosmetic symbolic link change for a README file. No result changes. Appreciated.

zach1221 commented:
@jkbk2004 @BrianCurtis-NOAA ok I'm going to restart the testing.

jkbk2004 commented:
@zach1221 @FernandoAndrade-NOAA @BrianCurtis-NOAA Finally, it's all updated ok. I think it's safe to re-do the tests on RDHPCS side.

BrianCurtis-NOAA commented:
> @zach1221 @FernandoAndrade-NOAA @BrianCurtis-NOAA Finally, it's all updated ok. I think it's safe to re-do the tests on RDHPCS side.

HYCOM-interface has a change but no .gitmodule entry. Please fix.

jkbk2004 commented:
> > @zach1221 @FernandoAndrade-NOAA @BrianCurtis-NOAA Finally, it's all updated ok. I think it's safe to re-do the tests on RDHPCS side.
> 
> HYCOM-interface has a change but no .gitmodule entry. Please fix.

HYCOM hash is already committed one on emc/develop. Good to go with hash update.

zach1221 commented:
@BrianCurtis-NOAA let us know how Acorn's going, if we should skip or wait.

zach1221 commented:
@jkbk2004 @FernandoAndrade-NOAA I talked to Brian offline. We're good to skip acorn.

DavidBurrows-NCO commented:
Thanks for everyone's help with this PR @jkbk2004 @zach1221 @FernandoAndrade-NOAA @BrianCurtis-NOAA @SamuelTrahanNOAA @RatkoVasic-NOAA @DusanJovic-NOAA @junwang-noaa @weihuang-jedi
